server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java

```diff
@@ -88,22 +98,29 @@ public ClientQuerySegmentWalker(
         }
     );
 
-    return new FluentQueryRunnerBuilder<>(toolChest)
-        .create(
-            new SetAndVerifyContextQueryRunner(
-                serverConfig,
-                new RetryQueryRunner<>(
-                    baseClientRunner,
-                    retryConfig,
-                    objectMapper
+    return new ResultLevelCachingQueryRunner<>(
```

Comment:
this feels like it violates the fluent workflow, is there a way to make this work inline with the fluent query runner?